### DISCUSSION TITLE
re-export of aws-smithy-types in dynamodb sdk

### DISCUSSION THREAD

**Discussion by pdeva** [awslabs/aws-sdk-rust#1127] [Q&A] [answered]
the [readme](https://github.com/smithy-lang/smithy-rs/blob/main/rust-runtime/aws-smithy-types/README.md) for `aws-smithy-types` says:

```
Generally, you should not need to take a direct dependency   
on this crate as service clients should publicly re-export the types when used.
```

However, in [AttributeValue](https://github.com/awslabs/aws-sdk-rust/blob/45ba2a808bb01f4875229acae73ca994bd75177e/sdk/dynamodb/src/types/_attribute_value.rs#L17), to pass a `Blob` object, one will indeed need to take a dependency on `aws-smithy-types`.

is this something that needs to be fixed in the dynamodb sdk or am i using it incorrectly somehow by depending on `aws-smithy-types` ?

**Comment by jdisanti** [comment, marked as answer]
Blob is in the `primitives` module in `aws-sdk-dynamodb`: https://docs.rs/aws-sdk-dynamodb/latest/aws_sdk_dynamodb/primitives/struct.Blob.html

**Reply by pdeva** [reply]
ah i see.

so i explored a bit more, and i notice that i dont see re-export of `::aws_smithy_runtime_api::client::result::SdkError` or `::aws_smithy_runtime_api::client::orchestrator::HttpResponse` anywhere.

![Screenshot 2024-04-16 at 10 21 33 AM](https://github.com/awslabs/aws-sdk-rust/assets/3190176/e00e9e7d-ab38-402b-b9ae-436983019965)



so for example, handling the result of [transactWriteItems.send()](https://github.com/awslabs/aws-sdk-rust/blob/45ba2a808bb01f4875229acae73ca994bd75177e/sdk/dynamodb/src/operation/transact_write_items/builders.rs#L8) cannot be done without depending on `aws-smithy-runtime-api` , which the [Readme for it](https://github.com/smithy-lang/smithy-rs/blob/6438a09bef34051b7ca4ac7bfded29f32cc6566f/rust-runtime/aws-smithy-runtime-api/README.md) also says shouldnt be needed.

**Reply by jdisanti** [reply]
SdkError is re-exported: https://docs.rs/aws-sdk-dynamodb/latest/aws_sdk_dynamodb/error/type.SdkError.html

Make sure you're searching the crate's docs in docs.rs for these. They should come right up.

What do you need `HttpResponse` for? It should only be needed if you're writing interceptors or other special use-cases.

**Reply by pdeva** [reply]
got it. yeah i think `HttpResponse` was just used cause it was part of the result generic type. i managed to get rid of it.
